### PR TITLE
Allocate Freebies, subscribe users to Stripe plan when freebies are exhausted.

### PR DESCRIPTION
**Pull request by barberj**
Keeping subscription.active? (current_period_end) up to date will be handled by bounty 69 (Stripe webhooks). _The webhooks should call subscription.refresh_stripe_customer_.

Bounty: https://assembly.com/signupsumo/bounties/68

### CODE REVIEW

File: app/controllers/subscriptions_controller.rb [hunk 1]
```diff
@@ -2,8 +2,7 @@ class SubscriptionsController < ApplicationController
   before_action :authenticate_user!
 
   def show
-    @subscription = current_user.subscription
-    @subscription ||= Subscription.new
+    @subscription = current_user.subscription || Subscription.new
```
Comment (vanstee) [inline app/controllers/subscriptions_controller.rb]: :+1:

File: app/views/subscriptions/_new_customer.html.erb [hunk 2]
```diff
@@ -97,19 +97,19 @@
...
                 <label>Exp Month</label>
-                <input type="text" class="form-control input-lg" placeholder="<%= Date.today.strftime('%m') %>" data-stripe="exp-month"> 
+                <input type="text" class="form-control input-lg" placeholder="<%= Date.today.strftime('%m') %>" data-stripe="exp-month">
               </div>
             </div>
             <div class="col-xs-4">
               <div class="form-group">
                 <label>Exp Year</label>
-                <input type="text" class="form-control input-lg" placeholder="<%= Date.today.strftime('%Y') %>" data-stripe="exp-year"> 
+                <input type="text" class="form-control input-lg" placeholder="<%= Date.today.strftime('%Y') %>" data-stripe="exp-year">
```
Comment (vanstee) [inline app/views/subscriptions/_new_customer.html.erb]: Thanks for cleaning up after me :)

File: app/models/free_signup.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class FreeSignup < Signup
+  before_create :update_freebie_count
+
+private
+
+  def update_freebie_count
+    self.user.update(freebie_count: self.user.freebie_count-1) if influential
+  end
+end
```
Comment (vanstee) [inline app/models/free_signup.rb]: Ah nice. I like this whole pattern of always having a subscription. Good call.

**Comment by vanstee** [comment]
Looks good. Is there anything I need to do to make sure our existing users have the correct signup counts and subscriptions after this migration?

**Comment by barberj** [comment]
So nothing really needs to be done... if they already have paid subscriptions they could just get 110 subscriptions in a month (the free will be used up before the 100 are counted against the paid).

I'm working some more specs, but i could add those into bounty 69 work since the webhooks will be necessary to keep the current_period_end up to date.

**Comment by vanstee** [comment]
Nice yeah I'll go ahead and merge this stuff and close it out.

**Pull request merged**